Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Quality/QualityCVarGroup.h>
+#include <AzFramework/Quality/QualitySystemBus.h>
+
+#include <AzCore/Settings/SettingsRegistryImpl.h>
+#include <AzCore/Settings/SettingsRegistryVisitorUtils.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <AzCore/Console/IConsole.h>
+#include <AzCore/StringFunc/StringFunc.h>
+
+namespace AzFramework
+{
+    QualityCVarGroup::QualityCVarGroup(AZ::SettingsRegistryInterface* registry, AZ::IConsole* console, AZStd::string_view name, AZStd::string_view path)
+        : m_console(console)
+        , m_numQualityLevels(0)
+        , m_path(path)
+        , m_qualityLevel(0)

```

Comment:
Should this be initialized to zero or it should be `LevelFromDeviceRules`?

Reply:
Here I was thinking, what if I'm on a platform where I don't want or need to use defaults or device rules, should the user be able to provide a single quality group with all the settings in it and not specify a default level if there only is one level?

Reply:
Well is 0 always the first quality level?

In the case where the number of quality levels are 0, `m_qualityLevel` should be unset in a kind of way.
Like it should be some kind of "optional" construct that is conditionally set if m_numQualityLevels is > 0.
Like some kind of class that can wrap an "optional" value, but is "standard" to o3de"

Reply:
If no quality levels are defined then that is currently an error and will fail a check on line 115, I was thinking of the case where there is a single quality level defined which is likely the case on some platforms where the hardware is always known - though I suppose it's possible developers want to use quality levels to let players adjust settings on all platforms even if they don't need device rules.

Reply:
I went ahead and defaulted to `LevelFromDeviceRules` for now till we have a better idea about whether no default is case we should support.